components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp

```diff
@@ -878,5 +880,182 @@ namespace scream {
             });
         }
 
+        void compute_aerocom_cloudtop(
+            int ncol, int nlay, const real2d &tmid, const real2d &pmid,
+            const real2d &p_del, const real2d &z_del, const real2d &qc,
+            const real2d &qi, const real2d &cldfrac_tot, const real2d &nc,
+            real1d &T_mid_at_cldtop, real1d &p_mid_at_cldtop,
+            real1d &cldfrac_ice_at_cldtop, real1d &cldfrac_liq_at_cldtop,
+            real1d &cldfrac_tot_at_cldtop, real1d &nc_at_cldtop) {
+            /* Summary of algorithm
+            The goal of this routine is to calculate properties at cloud top
+            based on the AeroCOM recommendation. There are two main parts of
+            the algorithm: probabilistically  determining "cloud top" and then
+            "calculating properties" at said cloud top.
+            */
+
+            /* PART I: Probabilistically  determining cloud top
+            We treat model columns independently, so we loop over all columns
+            in parallel. We then loop over all layers in serial (due to
+            needing an accumulative product), starting at 2 (second highest)
+            layer because the highest is assumed to have no clouds. Let's take a
+            photonic approach from above the model top. Let's say pp(k) is
+            the probability of a photon passing through the layer k. We follow
```

Comment:
This also assumes that cloudy portions of the layers are completely opaque, right? Because technically I think you'd have a non-zero probability of a photon passing through even cloudy layers.